src/schema/posts.ts

```diff
@@ -741,11 +742,31 @@ export const resolvers: IResolvers<any, Context> = {
       return { _: true };
     },
     deletePost: async (
-      source,
+      _,
       { id }: { id: string },
       ctx: Context,
     ): Promise<GQLEmptyResponse> => {
-      await ctx.getRepository(Post).update({ id }, { deleted: true });
+      if (ctx.roles.includes(Roles.Moderator)) {
+        await ctx.getRepository(Post).update({ id }, { deleted: true });
+        return { _: true };
+      }
```

Comment:
Why do we manually check a role here, should that not be handled by ensureSourcePermissions?

Reply:
This is for an app-wide Moderator role. Like banning any article, etc.